CommunityToolkit.Mvvm/ComponentModel/Attributes/AlsoValidatePropertyAttribute.cs

```diff
@@ -0,0 +1,44 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace CommunityToolkit.Mvvm.ComponentModel;
+
+/// <summary>
+/// An attribute that can be used to support <see cref="ObservablePropertyAttribute"/> in generated properties, when applied to
+/// fields contained in a type that is inheriting from <see cref="ObservableValidator"/> and using any validation attributes.
+/// When this attribute is used, the generated property setter will also call <see cref="ObservableValidator.ValidateProperty(object?, string)"/>.
+/// This allows generated properties to opt-in into validation behavior without having to fallback into a full explicit observable property.
+/// <para>
+/// This attribute can be used as follows:
+/// <code>
+/// partial class MyViewModel : ObservableValidator
+/// {
+///     [ObservableProperty]
+///     [AlsoValidateProperty]
+///     [Required]
```

Comment:
Having it opt-in makes it consistent with `[AlsoBroadcastChange]`. Note that using this attribute is not mandatory: it only enables each setter to validate its property immediately. You might very well just want each property to be annotated but then to validate all of them together when eg. clicking a "submit" button, or something. This would enable you to support both scenarios just fine.

Reply:
I think the concern being raised here is that if you always want all the properties to be evaluated immediately and have a ton of properties on a class called `ObservableValidator` then it's a lot of extra annotations. Where broadcasting a property is going to be less common.

Should there be the ability to put this at the class level to make it apply to all properties underneath?

Reply:
I think this should be an opt-out scenario instead of opt-in, but an attribute at class level would be better than specifying this at every property.  I could see this attributed added to a base class and then the need to opt-out at specific properties.

Reply:
I really don't want things to get too complicated, plus I also really don't want to add more and more features now that we're so close to an 8.0 release, I really want a stable build to just go out so that everyone can use this. I think it might be fine to only provide opt-in, and then we can always eventually think about some opt-out system later on.

Reply:
I know, but I think the concern here is what the common scenario here is.

- Is it more common that folks want immediate feedback from all their properties as folks are filling out a form?

- Or is it more common that users only get feedback when clicking a 'submit' button?

Like those are the end-results of this choice from a UX perspective.

I'd argue that as a user it's more helpful if a form tells me I did something wrong right away compared to only at the end once I've filled everything out. Adding @niels9001 to help out too with this discussion.

In that case, we'd want an opt-out mechanism as the expectation/better scenario is to just evaluate on change by default, right?